substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/Target_java_lang_StringConcatHelper.java

```diff
@@ -37,6 +38,16 @@
      * replaces the allocation with a deopt.
      */
     @Substitute
+    @TargetElement(onlyWith = JDK14OrLater.class)
+    private static long checkOverflow(long len) {
+        if (len < 0) {
```

Comment:
You missed a subtle part of http://hg.openjdk.java.net/jdk/jdk/file/6bc1fbc6e006/src/java.base/share/classes/java/lang/StringConcatHelper.java#l54 : the conversion to `(int)` before the comparison.

This shows that my substitution-based approach for performance optimization was misguided. Please remove this whole file, we need to find a more maintainable way of doing the optimization.

Reply:
I've removed the file.  Not sure why this shows up as a rename.  I manually added JDK15OrEarlier.java??

Reply:
removed StringConcatHelper.